packages/analytics-browser/src/browser-client.ts

```diff
@@ -169,12 +169,11 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
     return this.config?.sessionId;
   }
 
-  setSessionId(sessionId: number) {
+  async setSessionId(sessionId: number, shouldTrackNewCampaign?: boolean) {
```

Comment:
Instead of making this `async` we could make it return a "Promise object" like we do for other methods (e.g. `await amplitude.track().promise`).

This pattern allows users to choose whether or not them want to await the result without needing to always explicitly ignore the promise with `void setSessionId()`

Reply:
Thanks the sugesstion. I return the `returnWrapper` in the setSessionId.